### PR TITLE
Policy object pusher and security group of interests converters

### PR DESCRIPTION
**Pull request by radkrawczyk**
# Pull Request summary:
Extracting policy object pusher from config pusher
Introducing UrlFiltering converter
Adding initial shape of  abstraction for v1->v2 references update

# Checklist:
- [x] Make sure to run pre-commit before committing changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

**Comment by radkrawczyk** [comment]
Push works correctly, clear_ux2 and rollback needs fixes. working on that

File: catalystwan/utils/config_migration/creators/policy_object_pusher.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+    @abstractmethod
+    def update_references(self):
+        pass
+
+    @abstractmethod
+    def are_references_update_required(self) -> bool:
+        pass
+
+    def get_v2_uuid_of_transferred_v1_policy(self, v1_uuid: UUID) -> UUID:
```
Comment (sbasan) [inline catalystwan/utils/config_migration/creators/policy_object_pusher.py:47]: can we use 'origin'/'target' to distinguish uuids (instead v1/v2)?

File: catalystwan/utils/config_migration/creators/policy_object_pusher.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    @property
+    def push_order(self) -> int:
+        return 2
+
+    def update_references(self):
+        if not self.are_references_update_required():
+            return
+
+        if advanced_malware_protection := self.transformed_parcel.parcel.advanced_malware_protection:
+            v2_uuid = self.get_v2_uuid_of_transferred_v1_policy(UUID(advanced_malware_protection.ref_id.value))
```
Comment (sbasan) [inline catalystwan/utils/config_migration/creators/policy_object_pusher.py:154]: can we use 'origin'/'target' to distinguish uuids (instead v1/v2)?

File: catalystwan/utils/config_migration/creators/policy_object_pusher.py (hunk 1 above)
Comment (sbasan) [inline catalystwan/utils/config_migration/creators/policy_object_pusher.py:47]:
```suggestion
    def get_target_uuid(self, origin_uuid: UUID) -> UUID:
```

**Review by sbasan — COMMENTED** [review]
Suggestion to use different naming

File: catalystwan/utils/config_migration/creators/policy_object_pusher.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+class ReferencesUpdater(ABC):
+    transformed_parcel: TransformedParcel
+    pushed_objects_map: Dict[UUID, UUID]
+
+    def get_parcel(self) -> AnyParcel:
+        return self.transformed_parcel.parcel
+
+    @property
+    @abstractmethod
+    def push_order(self) -> int:
```
Comment (sbasan) [inline catalystwan/utils/config_migration/creators/policy_object_pusher.py:36]: maybe push priority is better name as it is single number?

**Review by sbasan — APPROVED** [review]

**Pull request merged**